### DISCUSSION TITLE
Why does a custom button not show?

### DISCUSSION THREAD

**Discussion by DrDryg** [AvaloniaUI/Avalonia#12889] [General]
I just started to learn Avalonia and trying out some things. Like modifying examples that I can find. 

So I found an example where a Border is converted to a CustomBorder. In its simplest form it is just:

`public class CustomBorder : Border {}`

If I include this in my xaml file it works like a Border. No surprise for me.

If I do exactly the same thing for a Button it does not work:

`public class CustomButton : Button {}`

Then if I include it in my xaml file. The button is not shown in my window. 

Why is this?

**Comment by jp2masa** [comment]
That is explained in the info [here](https://docs.avaloniaui.net/docs/styling/selectors#oftype). If you want to apply `Button` styles to `CustomButton`, you can override `StyleKeyOverride` and return `typeof(Button)`.

**Reply by timunie** [reply]
More about custom controls: https://github.com/AvaloniaUI/Avalonia.Samples#%EF%B8%8F-custom-controls-samples

**Reply by samyraj42** [reply]
> That is explained in the info [here](https://docs.avaloniaui.net/docs/styling/selectors#oftype). If you want to apply `Button` styles to `CustomButton`, you can override `StyleKeyOverride` and return `typeof(Button)`.

Can you share examples? i also face the same issue.

**Reply by thevortexcloud** [reply]
There is examples in the docs.

https://docs.avaloniaui.net/docs/basics/user-interface/styling/styles#style-key

**Comment by DrDryg** [comment]
Thanks a lot, solved the issue. So if I get it right. No style of my CustomButton is set so I will not see anything. But then, what do I need to set in the style for it to become visible in some sensible way?

**Reply by maxkatz6** [reply]
If you want your button to use default button style, then follow @jp2masa advice and override StyleKeyOverride property. It will reuse all normal Button styles, but you still can add your own logic in your class.

If you want to implement custom styles for your custom button, you can see how it is done for the Button itself:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L13

That's a ControlTheme defined for Button type, and included in Application.Resources (through FluentTheme).

**Reply by maxkatz6** [reply]
See https://docs.avaloniaui.net/docs/next/basics/user-interface/styling/control-themes
